src/Bicep.Core/Analyzers/Linter/ApiVersions/ApiVersionProvider.cs

```diff
@@ -19,12 +18,11 @@ public class ApiVersionProvider : IApiVersionProvider
         // One cache per target scope type
         private readonly Dictionary<ResourceScope, ApiVersionCache> _caches = new();
         private readonly IFeatureProvider features;
-        private readonly IAzResourceTypeLoader resourceTypeLoader;
-
-        public ApiVersionProvider(IFeatureProvider features, IAzResourceTypeLoader resourceTypeLoader)
+        private readonly IEnumerable<ResourceTypeReference> resourceTypeReferences;
+        public ApiVersionProvider(IFeatureProvider features, IEnumerable<ResourceTypeReference> resourceTypeReferences)
```

Comment:
REMINDER for me to look at perf...

Reply:
I think this is laudable here but at the same time unsustainable @majastrz, @anthony-c-martin should we add a ticket to have a framework to test performance so we can apply consistently the necessary checks and also prevent unintended regressions?